### ISSUE TITLE
[no-unused-modules] unusedExports false positives in multi-line imports and multiple files

### ISSUE THREAD

**Issue by roikoren755** [benmosher/eslint-plugin-import#1745]
First of all, let me say how much I appreciate all the hard work that went into this plugin, and how helpful it is in all of my projects. Now, to the issue at hand.

If I turn on the `import/no-unused-modules` rule, with the `unusedExports` option set to `true`, I get false positives for unused exports when I have multi-line imports from that file via an interim file.

In my main repo, where this issue first happened, the use case is in a React application, a component directory has a `components` sub-directory, with an index file exporting different helper components from different directories within that directory. When I have too many components exported from that `components` sub-directory, I like to split the import statement into two (or more), instead of having a single long line, or several very short ones, each with a single name of a component imported (I use prettier in that repo as well). In such cases, ESLint warns me about unused exports from the second import statement and onwards.

To simplify matters, if for example I have in `foo.js`:

```
export const foo = 1;
export const bar = 'baz';
```

and in its sibling `bar.js`:

```
export { foo, bar } from './foo.js';
```

and finally in `baz.js`:

```
import { foo } from './foo.js';
import { bar } from './foo.js';

console.log(foo, bar);
```

ESLint warns me in line 1 of `bar.js` about unused exports - `exported declaration 'bar' not used within other modules  import/no-unused-modules`.

A reproduction can be found in https://github.com/roikoren755/eslint-plugin-import-bug-reproduction. To see the bug simply clone the repo, run `yarn` and `yarn lint`, and voila, an error message will appear, even though it shouldn't.

**Comment by roikoren755** [comment]
Just to drive the point home, if you change `baz.js` in the reproduction repo to import both `foo` and `bar` in the same statement, the error isn't reported anymore

**Comment by ljharb** [comment]
In `baz.js`, what happens if you consolidate those two imports into a single import statement, as is best practice?

**Comment by roikoren755** [comment]
As I said, consolidating the imports results in no errors reported. In this simple case, that is what I do, but what about the use case I described? Using `prettier` or the `max-len` rule from `eslint` will cause a long import to span several lines, and I'd prefer splitting the lines, though not if that will lead to errors from this rule

**Comment by ljharb** [comment]
I'd strongly discourage you from repeating import specifiers; an import spanning several lines is fine.

That said, this rule should be able to work with repeated specifiers.

**Comment by roikoren755** [comment]
> I'd strongly discourage you from repeating import specifiers; an import spanning several lines is fine.

If you don't mind, can you please elaborate on this? from what I could find, the imported module is only executed once, per the ES specification, so the only downside to multiple import statements should be a slightly larger compiled code. Why do you strongly discourage this?

**Comment by ljharb** [comment]
Not for a technical reason, but for a conceptual one - it makes code harder to maintain.

**Comment by roikoren755** [comment]
Got it, thanks :)
Also, if there is a way I could help with allowing the rule to work with repeated specifiers, let me know

**Comment by ljharb** [comment]
A PR would be helpful; at the very least, a PR with failing test cases :-)

**Comment by roikoren755** [comment]
I tried adding a failing test case, but could not figure out how to make the rule tester run lint in the context of a directory, taking into account three files in it.

Going over the source code, I think I narrowed the issue down to the right place - line 406 in `ExportMap.js`. In the static `.parse` method, for each import declaration found in a file, an entry is entered into the local instance of ExportMap's imports map. If the same remote path is imported from more than once, the second import declaration gets skipped altogether, with that function instead returning the `getter` computed for the previous import declaration.

![image](https://user-images.githubusercontent.com/26850796/82129113-1c76f480-97c9-11ea-8310-c1cceac707c7.png)

The following code fixes the issue in my reproduction repo, and still passes all tests, though I'm not sure how it will affect error reporting in import declarations after the first one.

![image](https://user-images.githubusercontent.com/26850796/82129249-06b5ff00-97ca-11ea-8284-12614ed21397.png)

I'll open a PR with this change, to get more comments and/or suggestions on better fixes

**Comment by ljharb** [comment]
Thanks, that's a great start.
